### ISSUE TITLE
notes on Apple QoL improvements

### ISSUE THREAD

**Issue by jimbrend** [jimbrend/notitle#1]
watchOS issue 

Current version,

Switching on Spotify 

For workarounds I will contact each company.

**Comment by jimbrend** [comment]
![Image 2-23-24 at 1 23 AM](https://github.com/jimbrend/notitle/assets/8608634/4ead1b78-8a88-4335-81cb-649cacb18bca)

The fruits of my labor, this above segments were submitted through the public portal and none of this is internal information, but my recommendations I may utilize in the future or work on little QoL apps,

 I'm writing them here as very sporadic notes I can come back to or ideas 

It may be nice to aggregate a Find My solution for what information is displayed when locking devices: https://support.apple.com/en-us/101558

For the above you may dox your personal information when leaving a note on a personal device if someone then looks up your name or e-mail correctly they get your full name still, when not realizing it instead of using an iCloud relay e-mail instead or simply keeping the default blank lock message as it is optional, information is confirmed to display if correctly guessing someone's e-mail such as full name last time I checked, these are some friction points I'm noticing that lead to people not following all the correct steps, or being uncertain what information is still accessible when they lock their device which should be nothing and show lost mode unless you can find your device and input your passcode, or require the device to be erased and then activation locked if the passcode is inputted incorrectly
https://support.apple.com/en-gb/HT204306

Watch passcode and Apple passcode also can be different, 

You can create additional iCloud e-mails, it is optional you can use any e-mail as your AppleID e-mail and link other e-mails, but the first iCloud e-mail you create is forever tied to your account.  You are stuck with it and can only change it if you ask for a name change as per current guidance,

It's the biggest uphill battle to more UX improvement such as these because obviously these impact so many users, but something people deal with the most daily in my experience, just typing information here on biggest improvements I think can be implemented:

- Increased redundancy when creating a recovery key, you cannot recover an account if you generate one and lose it 

- One Page/Tab or a new app that allows you to manage your contingency plan in the event you lose your device all on one screen, if you must recover you must expect that you cannot access your account for the duration of account recovery, this period involves an initial 24-72 hour period then an additional automated waiting period of anywhere from days to a month (based on how much information you can verify, phone number is required and things like verifying card numbers may speed up the process of waiting on the second waiting period as Apple must establish a new way to contact you on alternative recovery options you provide after verifying at iforgot.apple.com): https://support.apple.com/en-us/HT204921, this is often miscommunicated because you must first again wait 24-72 hours, then you can get a visual of the number of days exactly left to regain access at iforgot.apple.com in the next automated wait period no human being is able to help with - which takes days to ~month based on verifiable information, this was also not well communicated and any device logged in still and powered on resets the waiting time and you must submit an account recovery request again (something as simple as leaving your Apple Watch on after you've waited a few days and if it turns on, it will reset the process, and you must resubmit an account recovery and continue to wait).  I document more things that you can do [here](https://x.com/jimsbr/status/2003648914713411883?s=20) in a recent overview of this, but most of it is in the [Apple Support app officially maintained by Apple](https://apps.apple.com/us/app/apple-support/id1130498044)  (which is not by default easily shown anywhere on the Home Screen or Settings screen to start, recommending best practices such as first trying on another device you are logged into to regain access).

This is excluding any process that VIPs may have access to in a separate VIP queue handled by historically Cupertino based senior and executive team members for government officials, even athletes and their relatives as publically documented, or those who are usually recommended to remain in lockdown mode on their phone to avoid other issues sometimes-- which is usually unnecessary and convoluted to explain at all.
 
 A contingency plan well documented on accessible devices would also list all your non-Apple apps that currently rely on your phone number or access to your device preferably on a local machine, an easy way to regain access is using the Apple Support app on another Apple device that is available such as a family or friend and if you input passcode correctly and details you can regain access quicker (you go to the Apple Support Apple on their phone or even one at the store, you can then go to recover, if the demo device or the device is already logged in you can tap on an option that says recover another account while that account remains logged it in fact speeds up the process by verifying your Apple Passcode on another apple device's system faster), sometimes people are unable to get to a device and having to buy a burner phone and wait or creating another account is often inconvenient as the device turns into a paper weight showing just the domain name of the e-mail during Activation Lock, only the account owner removes it-- if your account is fully compromised for any period of time by someone else that is an issue, and you must go to icloud.com/find to lock devices appropriately, you must know your details to login.  

_(I forget if I mentioned already but you also MUST know your phone number in full for most modern accounts with by default authentication enabled.  You can even test this yourself by going to iforgot.apple.com and testing the process to begin you will see exactly what # it needs it will show the last two digits of the numbers.  It may even be an older number or if you have added several in your communication preferences, if you don't know what it is you are out of luck here and cannot proceed.  This requires you to go through your old records to either remember or identify what your old number is or you will not be able to access that account again if you get locked out, this is the same thing if someone has compromised your account and changed your details you must make sure ideally I usually suggest you have a device you are still logged into somewhere such as an extra iPhone or iPad available... You can call 1-800-My-Apple if you have further questions they can even screen share with you display arrows on your screen and direct you to where to update your preferences or to address any concerns... For legacy accounts that are much older usually that don't have authentication you need to know the birth date you entered as well and verify additional information)._
    
- Users who want to backup private photos have immense difficulty navigating these myriad of articles:
https://support.apple.com/en-us/108782
https://support.apple.com/en-sg/HT204215
https://support.apple.com/en-us/108429
https://support.apple.com/guide/icloud/introduction-to-icloud-mm74e822f6de/icloud#:~:text=When%20you%20use%20iCloud%2C%20you,same%20photos%20on%20both%20devices.

- It would benefit to feature on iCloud.com/photos you can click and drag 1000s photos at at time to local storage, or find a workaround to be able to do your entire iCloud via the web easily (this is an easy click and drag method but limited to the 1000), the biggest use-case I've seen is users want to transfer their most precious moments and save them easily in a family archive or external drive by plugging quickly and clicking and dragging in the least amount of steps possible, the UX on this is bad for non-tech savvy users and often leads to lost data
https://support.apple.com/en-in/111762

i.e. grandma expects to be able to plug in a family drive then simply click and drag all photos easily to it an external storage device from one modal

There are several related ways to do this linked above for data transfer but it's hard to describe to users the nuances of the Apple ecosystem where data lives, based on all the support documentation I describe it is located in one of three places:

- General > Device Storage is your local storage on your physical device
- iCloud > is your iCloud Storage
- iCloud > In your Backup which is not accessible unless you wipe and restore on the same or another compatible device already updated to the correct version

- Conveying the fact to users is a recurring challenge that your backup will only have your last successful backup if kept turned on, your iCloud Storage must also be synced and turned on, and your local storage is gone if your device is obviously broken or lost,

- Many users may have one iPad for instance not syncing and not realize, or there is a gap in understanding of what is syncing to the cloud or on the iPhone itself, 

This leads to issues of losing precious old text messages users cherish or need for instance and they get lost and those require more steps, conveying to a user you have to go to iCloud and try and restore multiple different text message backups is an issue sometimes too: https://support.apple.com/guide/iphone/recover-deleted-messages-iph16ecebf48/ios

Ideas for companion apps or improvements:
- One solution could be a new iCloud interface or interface that boots up and allows a simpler click and drag experience and one tab that shows the data stored in each location and the exact discrepancies between them including the total storage in your backup and on each device all on one screen in an easily digestible graph, and destinations on right tab

- The contingency plan app showing all information on one modal, including all third-party accounts reliant on your phone number or iCloud account without tapping through multiple screens. 

- This would negate all the years of Android users' ability to click and drag files easier after one plugin instead of users' having to sometimes guess and then attempting to learn where year old albums are in and relieve lots of stress during any crisis

I worked for Apple for six years aggregate including in-house at corporate not just as a contractor (I have also triple-checked this all of this is public information to be helpful for users or app builders to help with QoL things), I don't want to write more about any negativity that is subjective... I resigned my position to be clear communicatively from my current role when receiving a higher offer thus far, and have cleared I am eligible for rehire and provide nicer anecdotes at the end of this.  I had maintained and as you can see this repo has stayed live and not taken down on GitHub as a reference (while other repos such as clones of the App Store's source code have been dealt with promptly and are unavailable in less than 24-48 hours)... These common problems not being addressed on a timeline- for a company with 10s of thousands of employees and a market cap that broke two trillion while I was there then three trillion (currently as I edit this sitting at #2 Market Cap just behind NVIDIA only) seem kind of not cogent or sometimes frustrating like for more simple things even I run into still like switching between or updating billing to download apps for many that do not auto-pay on multiple subscriptions. 

Even recently, I spoke with Apple to sort that last problem.  Between five representatives over the course of two-three months I received no clear answer, no investigative response, no solution, and contradictory answers.  Finally, I went into the screen-share we looked, this is in fact a problem that is not solvable, and is not clearly shown to the user.  I can look at view account and see charges, I can look at subscriptions and see charges all cleared and non-pending.  I can go do to download an application and link any card I want with an infinite amount of money on it clearly as a Bank Debit Card or Credit Card from a reputable non-prepaid non-neo bank ready to be charged.  I can not know without checking two-three screens, nor know definitively how much the card would be charged again.  I tried it just last month after updating this article as I write this Jan 31st, 2026... I tried it, I checked everywhere, I confirmed what would charge me.  It charged me for two other things that did not update until after the fact waiting only a few minutes, only after it would update correctly across these screens, confirmed by Apple representative.  There is too many insurmountable UX corporate hurdles to leap through, should I advocate for a clear update like this, among dozens and dozens.  

Again, I hope these Quality of Life notes inform others.  I am still in awe of how amazing these devices are, how much they improve our lives, even recently I remember discussing this with a current Apple employee we were discussing the history of the company we both (me moreso becoming a bigger fan of Apple after Apple Silicon), how much we like this company and the product lineups.  There are literal 1,000s of people working on these issues as they arise exactly as I have documented them here.  Everyday usually as I would check and be informed on anywhere from 2,500 - 5,000 actively available on the clock to 8-11,000 employees total working on these things.  Even site managers and corporate alike most log their time spent, even teams behind the cloaked teams working on special projects in corporate and more log their time and must switch it (similarly to roles that use Salesforce or other time trackers) to log when you are on lunch or working on a project/are in training etc.)  The total contact of this recurring things I describe below account for sometimes 200,000 contacts between e-mails, bugs reported, active calls, or issues in just one hour.  I have not seen clear explanations documented by anyone on many of them yet in fact, and when I search for common issues I find nearly five to eight year old Apple discussion threads, that really must be updated in SEO...  There have been agentic internal tools implemented on the proprietary software suites though, that do make instantly identifying common resolutions to pop up easier now, with slow rollouts when working on certain problems (also simply because of the sheer scale of users impacted many things must be considered to keep the UI so user-friendly)... but a resolution to the problem that reduces even first-contact must gain traction within the corporate stakeholder environment with increased attention too...

I am neutrally recording this and here transcribed just a fraction from my own notes as well and work (excluding so many setup processes, and across all other devices, and I have not shared any internal information or sensitive data PII or anyting, and thus this repo is still live of course, thankfully, and would like to continue to use it, as a reference)...  If there are any Apple employees that would like to give me any further feedback on what to leave up or take down, please share with me.

I am also a fan from what I've seen of Apple's privacy, I believe I had access to most of anything Apple had access to in a user's account, I even on occasion several times did liaison directly with law enforcement contacted on different channels directly from their departments who questioned me about accounts.  I never was able to see the location of someone's device, or any data on that device or in their account.  Only you as the user can access your Find My, and Apple representatives can help you see the exact same numbers populated simply with how much data is in storage only (i.e. how is the # of GBs stored not what is stored to my knowledge).  This was not necessarily the case and as stringent at other companies, and I could only see accounts I was actively working on, not even allowed to pull up my own account and I respected that rule and it was enforced.  Only I've seen it enforced as much at the cable company (who truly have more capability and at wireless carriers since Apple is not a wireless carrier to triangulate or provide location data from the cell towers i.e. also employees trying to look up a celebrity's account out of curiosity and being logged for that is a first-time fireable offense), I would say that should also be upheld in a lot of other environments e.g. log anytime an employee account interacts with any user's account.  Above I shared how you can restore in a separate process in your own iCloud.com web interface too your own iCloud messages to different dates they were backed up and restored separately too.  However, SMS or even calls/voicemails you may need access to and lost are now either lost forever, that data exists somewhere else not on Apple's servers, or the metadata with the carriers themselves.  I can confirm that I worked on high-profile cases involving lawsuits in the hundreds of thousands too where a user needed access to their messages and confirmed as such (this was a normal thing done sometimes among many other numerous tasks I'd complete before noon, and I remember that one because my office colleagues around me still mostly went to lunch on time)... Again, a lot of this fragmented and already well-documented as far as what is simply the current processes in place to address these issues inevitably arising (for example on Apple forums too where when you Google some of these problems you are led to community answers instead of official answers with SEO and no updates to the original articles that constantly too are still appearing in search and in new LLM tools and referenced by others erroneously).  

Some of these posts appear or are referenced and have Apple employee answers too and are not updated or refer to current help documentation.  This leads people to think, maybe there is no solution to their problem, they lose access to precious data, or take some other route that is very non-optimal, wasting their time, money, and energy.

Furthermore, the M chips did make me more of an Apple fanboy when I previously was not (usually building my own Windows machine... but still there were many other life reasons I gave up this desk badge for now, unfortunately)...

![9F068E26-1A2C-46BB-955C-5143DABBD1A2_4_5005_c](https://github.com/user-attachments/assets/ee3c8b7e-bd92-4063-a58a-69953a30bca2)


A few other improvements could be made like if I am about to download an app it says click billing info then nothing happens even if you are all paid out on all subscriptions, you login you click billing info, nothing happens- does not open your browser on Mac Sanoma 4.1.2... then, Apple must address 200,000 contacts total in a day across all channels on several complex issues, billions of interactions or microinteractions total and people get directed to the Media Department as an App Store issue, vice versa, or are misdirected even within the UI still for certain problems (maybe the public simply declaring them moot points, and settling into the ecosystem)... Things of course should continue to be clear, what is a wireless carrier and what is an Apple issue.  As Apple is not a wireless carrier, and Apple is indeed not responsible for how exactly each third-party application operates, since they are not maintaining in-house support for any third-party non-Apple application, like Snapchat, Facebook, or Instagram.  If you are IP banned or device banned or however locked out of one of those applications, Apple cannot help you with your other accounts.  

Another issue is if one app was a paid app and you recover, you had an old e-mail on that app, you have to go through all your billing history to resolve to figure out what old e-mail you have to log into.  At the time, the only solutions were to start anew, or delete apps one by one (the entirety of all the applications on your phone), cross referencing with what you remember the app was and attempt again after each one, regain access to the old e-mail, or you keep having to tap the message to ignore it until resolved... for paid apps on old e-mails, such as significant others sharing a device.

I'd rather document to help others maybe resolve the UX in some way somewhere, then maybe make a video on it later with more detail here to expound on substantially at some point.  It's expected of a large corporate environment though, there are a lot of moving parts to deliver such a great experience of seamless usage of an awesome product lineup.  Or, one day maybe collaborate with Apple again (I can maintain a developer account now as I was told I could not have one as an employee), things they're doing in the computer architecture world across the world like even their buildings and offices in Cork, Ireland are still inspiring, and most development teams still ship their senior engineers Macs and Apple is still looked at as the bar of excellence... which I understand, but I'm also a gamer and I use Windows more sometimes, despite I view Apple as a productivity device sometimes moreso than a gaming device... or just all the things easier to access in one place.  Did not mean to add these last paragraphs but here we are, enjoi!
